### ISSUE TITLE
Improve Sorting of Events

### ISSUE THREAD

**Issue by hauf-toni** [digitalfabrik/integreat-app#2986]
**Is your feature request related to a problem? Please describe.**
the current sorting system for events causes long-term (ongoing) events to appear prominently at the top of the list. This reduces the visibility of single (one-time) events, which may be less noticeable to users. As a result, users may miss out on important individual events, leading to a less optimal user experience

**Describe the solution you'd like**
An improved sorting mechanism where single (one-time) events are prioritized and displayed at the top of the feed, followed by long-term (ongoing) events. This adjustment would ensure better visibility for time-specific events, making it easier for users to discover and engage with them

**Describe alternatives you've considered**
An alternative solution could involve visually differentiating long-term events with distinct labels, making them easier to distinguish. However, this approach might not fully resolve the issue of reduced visibility for single events

**Additional context**
The implementation is dependent on the backend release of the option to categorize events in single events and ongoing events (📆 [#3070](https://github.com/digitalfabrik/integreat-cms/issues/3070))

**Comment by steffenkleinle** [comment]
> An improved sorting mechanism where single (one-time) events are prioritized and displayed at the top of the feed, followed by long-term (ongoing) events. This adjustment would ensure better visibility for time-specific events, making it easier for users to discover and engage with them

@hauf-toni do you mean by this that all single events are shown first? And all long terms are shown below?
Should we add some visual separation and/or explanation why there are two different list (as it won't follow chronological sorting anymore)?

**Comment by hauf-toni** [comment]
I would suggest we discuss this together with @osmers in our upcoming meeting next week @steffenkleinle

**Comment by steffenkleinle** [comment]
@hauf-toni can you remember what decision we reached here?

**Comment by steffenkleinle** [comment]
> [@steffenkleinle](https://github.com/steffenkleinle) I don't quite remember but my proposal would be to implement a filter like "frequency" (see screenshot) and show a default sorting that shows single events first in chronological order and then show ongoing events. this ensures that urgent + time-specific events are visible and don’t get lost even before a filter is applied. we should then show ongoing events grouped below sorted by start date (or relevance but I'm not quite sure how that would be possible).
> <img alt="Image" width="307" height="657" src="https://private-user-images.githubusercontent.com/123894307/485572897-014d59ad-70bc-4025-80db-2c2106b7ca18.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTczMzIwMzQsIm5iZiI6MTc1NzMzMTczNCwicGF0aCI6Ii8xMjM4OTQzMDcvNDg1NTcyODk3LTAxNGQ1OWFkLTcwYmMtNDAyNS04MGRiLTJjMjEwNmI3Y2ExOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwOFQxMTQyMTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYTFmMzIzYzczYjA5NmFjNmY5NDViNGFhYmEzYzc2YzdjMjMwY2NmMTc5ZTJmNzI2N2EwZDJhZTFjNTAwODE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.VNsADdk5iHMagEPMvNBHvmJsQ3HxYwvByhVr3jcKx6A">
> 
> [Design here](https://www.figma.com/design/cA4F2MwHs2LNGviOWUjfE9/Integreat-Tickets-Frontend?node-id=5307-1952&t=L54yjUn5plaizfbh-1) let me know what you think.

I think it was something of a mixed logic like "Show the next three one-time events highlighted and then all events the same way before" or "Show all one-time events the next 7 days and then the other events like before".

But now that I think of it, what do you think of grouping events in general like:
- Today
- Tomorrow
- This week (i.e. next 7 days)
- This month (i.e. next 30 days)
- Further

In each of these groups, we could then show one-time events first and recurring evens after. I think this would make it a little clearer for users on when the event is roughly happening as well as improving the visibility of one-time events without adding recurring events to the very end.

Regarding the design with the filter: I think we could do that on top, but if you can only choose `One-time` and `Recurring`, the same ButtonGroup component as for "All Events" vs "New" would perhaps make more sense?
https://mui.com/material-ui/react-button-group/

**Comment by steffenkleinle** [comment]
@lunars97 I added a new todo based on https://github.com/digitalfabrik/integreat-app/issues/3689#issuecomment-4729140561: Remove the icons for today and today-recurring as its duplicated information with the groups

**Comment by lunars97** [comment]
> [@lunars97](https://github.com/lunars97) I added a new todo based on [#3689 (comment)](https://github.com/digitalfabrik/integreat-app/issues/3689#issuecomment-4729140561): Remove the icons for today and today-recurring as its duplicated information with the groups

Thanks a lot for informing, because I was about to continue working on it 👍 💚

**Comment by steffenkleinle** [comment]
@ulyanahoy-cloud @hauf-toni one question regarding this week/this month: It each refers to the next 7/30 days, not the literal week/month (i.e. until Sunday/until e.g. end of June), right? Do you think the naming `This week`/`This month` is clear enough here? For me it implies the latter a bit. Maybe it also doesn't matter though, just wanted to hear your opinion.

**Comment by ulyanahoy-cloud** [comment]
I agree that _This week / This month_ could be interpreted as the current calendar week/month. Maybe _Next 7 days_ and _Next 30 days_ would be a bit clearer.
